modules/super-admin/components/product-listing/ProductListingTable.tsx

```diff
@@ -9,105 +7,77 @@ import Button from '@ui/Button';
+  date_added: string;
+}
+
+const ProductListingTable: React.FC = () => {
+  // State to hold the fetched data from the endpoint
+  const [data, setData] = useState<ProductInfo[]>([]);
+  // State for search input value
+  const [searchVal, setSearchVal] = useState<string>('');
+  // State to hold the filtered products based on search and sorting
+  const [filteredProducts, setFilteredProducts] = useState<ExtractedProductInfo[]>([]);
+
+  useEffect(() => {
+    // Fetch data from the endpoint when the component mounts
+    const fetchData = async () => {
+      try {
+        const response = await fetch('https://test-super-admin.onrender.com/api/product/get-all-products/');
+        const result = await response.json();
+
+        // Ensure result is an array
+        const resultArray = Array.isArray(result) ? result : [];
+
+        setData(resultArray);
+
+        // Extracting specific fields: product_id, product_name, status, vendor_name
+        const extractedData: ExtractedProductInfo[] = resultArray.map((product: ProductInfo) => ({
+          product_id: product.product_id,
+          product_name: product.product_name,
+          status: product.status,
+          vendor_name: product.vendor_name,
+          date_added: product.date_added,
+        }));
+
+        setFilteredProducts(extractedData);
+
+        // Log the extracted data
+        console.log('Extracted Data:', extractedData);
+      } catch (error) {
+        console.error('Error fetching data:', error);
+      }
+    };
+
+    fetchData();
+  }, []);
 
   const handleSearch = (searchText: string) => {
-    const filteredProduct: Array<{
-      productName: string;
-      vendor: string;
-      id: number;
-      dateAdded: string;
-      status: string;
-    }> = data.filter((product) => product.productName.toLowerCase().includes(searchText.toLowerCase()));
+    // Filter products based on the search text
+    const filteredProduct = data.filter((product) =>
+      product.product_name.toLowerCase().includes(searchText.toLowerCase()),
+    );
     setSearchVal(searchText);
     setFilteredProducts(filteredProduct);
   };
 
   const handleFilter = (status: string) => {
-    const sortedProducts = [...data].sort((a, b) => {
-      const dateA = new Date(a.dateAdded);
-      const dateB = new Date(b.dateAdded);
-
-      if (status === 'newest') {
-        return dateB.getTime() - dateA.getTime(); // Newest to oldest
-      } else if (status === 'oldest') {
-        return dateA.getTime() - dateB.getTime(); // Oldest to newest
-      } else {
-        const statusOrder: { [key: string]: number } = {
-          Active: 1,
-          Sanctioned: 2,
-          Deleted: 3,
-        };
-        return statusOrder[a.status] - statusOrder[b.status];
-        // });
-      }
+    const sortedProducts = [...filteredProducts].sort((a, b) => {
+      const dateA: any = new Date(a.date_added);
+      const dateB: any = new Date(b.date_added);
+
+      return dateA - dateB;
```

Comment:
This is very error-prone sir, the new Date method returns a Date object in some case and this javascript arithmetic don't will throw an error when performed on objects. To successfully achieve ur objective, do it this way:

const dateA: number = new Date(a.date_added).getTime();
const dateB: number = new Date(b.date_added).getTime();

return new Date(dateA - dateB)